### ISSUE TITLE
Port forwarding doesn't work

### ISSUE THREAD

**Issue by krzikalla** [microsoft/vscode-remote-release#6656]
- VSCode Version: 1.66.2
- Local OS Version: Windows 
- Remote OS Version: Ubuntu 20.04
- Remote Extension/Connection Type: SSH
- Log (Remote Server):

[remoteagent] [error] Error: connect ECONNREFUSED 127.0.0.1:3000
    at TCPConnectWrap.afterConnect [as oncomplete]

- Log (Windows):

[renderer4] [error] read ECONNRESET: Error: read ECONNRESET
	at TCP.onStreamRead (node:internal/stream_base_commons:220:20)

Steps to Reproduce:

1. Forward port 3000 in the "Port" tab
2. Try to open localhost:3000 in Firefox (either via Ctrl-Click or via direct entering of the address)

There is no connection, only the above log messages appear. 
It had worked, altough I can't specify the exact point in time, when it stopped working.

**Comment by alexr00** [comment]
I'm not able to reproduce this issue. Can you still reproduce this issue with a minimal example (for instance, by running this server: https://github.com/microsoft/vscode-remote-try-node/blob/main/server.js)?

**Comment by krzikalla** [comment]
I've run this python server: https://flask.palletsprojects.com/en/2.1.x/quickstart/#a-minimal-application

and it worked (started on remote, forwarded port 5000 in VS Code Client, opens localhost:5000 in Firefox).
I am not sure what to do with your example, I may not run a .js file on remote.

I've tried on different remote machines. Coworker also tried on his machine. All with same result: ECONNREFUSED 
Trying other port numbers has no effect too.

**Comment by alexr00** [comment]
Tried it out with https://github.com/microsoft/vscode-remote-try-python so that I could test out a known working python server. 

Everything looked good there. Can you try out the server in that example? 

Can you share any `ssh` and `remote` settings you have set in VS Code?

**Comment by leeopop** [comment]
Unfortunately, I have the same problem. It seems working sometimes, but soon becomes not being able to connect to the forwarded port.

What I found is

1) It seems it only occurs in the Windows platform.
2) After system reboot, it starts working at the first time.
3) All browsers are not available to connect : It does not seem to be a browser-specific problem (such as DNS cache...).
4) Changing the port number was not a solution.
5) The problem starts occur from my desktop to my labtops one by one. I'm not sure it is due to Windows' auto update or VSCode's auto update. It seems a certain version of windows or vscode making this problem.

**Comment by kristiandupont** [comment]
I am experiencing this as well. My setup is running Linux in a VM on a Mac (m1). Port forwarding works most of the time, but every now and then it just stops. I can often restart vscode and make it work again but sometimes even that fails. I am at a loss.

**Comment by nathanrpage97** [comment]
@krzikalla Try on Firefox (127.0.0.1:3000) as a workaround.

I am using sphinx auto-build right now and I cannot access via `localhost:5000` but `127.0.0.1:5000` works. I can use live server just fine with `localhost:5000`. When I run sphinx-autobuild locally (on the ssh'ed machine) I can access just fine via localhost. Is this issue related?

logs from (Remote Server)
```
[2022-07-15 21:34:02.113] [remoteagent] [error] Error: connect ECONNREFUSED 127.0.0.1:5000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1161:16)
```

Remote machine is Ubuntu 21.04, local is Intel Mac 2019.

**Comment by krzikalla** [comment]
The problem still exists, but I can't add more information.  The error message in "Log(Remote Server)" is still:

[error] Error: connect ECONNREFUSED 127.0.0.1:3000
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1157:16)

**Comment by alexr00** [comment]
Tried it out with https://github.com/microsoft/vscode-remote-try-python so that I could test out a known working python server.

Everything looked good there. Can you try out the server in that example?

Can you share any ssh and remote settings you have set in VS Code?

**Comment by krzikalla** [comment]
Should have added in my former response, that neither installing Docker nor CodeSpaces is an option for me.
Regarding the settings all is on default as far as I can tell. Do you look for something specific?

**Comment by alexr00** [comment]
Any settings related to remote or ssh. Sharing your whole settings.json file would also be fine.

If you just make a minimal python server, similar to the one in https://github.com/microsoft/vscode-remote-try-python does it work?

**Comment by krzikalla** [comment]
OK, I got it. There was a fundamental misunderstanding on my side: before the port forwarding feature was added to VS Code I worked with a LiveServer extension. Then port forwarding became a VS Code feature and I thought I don't need a LiveServer extension anymore. Thus I desabled it. But then there was no one listening on remote. So now I have reactivated the LiveServer and in addition I have to forward the port in VS Code. Then things are working as expected.
